### PR TITLE
Telfe 277 floating panel

### PR DESCRIPTION
**Pull request by AlecsMissangu**
Component state are being internally for now. The UX is handled in the DS instead of outside for now. When there is a need to manage the state the outside the DS then we can extend the component

### CODE REVIEW

File: src/v1/components/surfaces/FloatingPanel/DraggableFloatingPanel.tsx [hunk 1]
```diff
@@ -37,8 +43,19 @@ const DraggableFloatingPanel: React.FC<DraggableFloatingPanelProps> = ({
       bounds={bounds}
       enableResizing={false}
       dragHandleClassName="drag-handle"
+      default={{ x: defaultPosition.x, y: defaultPosition.y, width: 'auto', height: 'auto' }}
+      css={{
```
Comment (ACoolmanTelicent) [inline src/v1/components/surfaces/FloatingPanel/DraggableFloatingPanel.tsx:47]: can we use sx?

I'm not particualrly worried - strictly speaking I'm not worried about this specific use

But if posssible I'd rather not leave `css` props around for people to copy and expand usage

File: src/v1/components/surfaces/FloatingPanel/FloatingPanelProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import React, { createContext, useState } from "react";
+
+interface PanelState {
+  visible: boolean;
+  minimised: boolean;
+}
+
+interface PanelsType extends Record<string, PanelState> {}
```
Comment (ACoolmanTelicent) [inline src/v1/components/surfaces/FloatingPanel/FloatingPanelProvider.tsx:8]: Suggestion: `[pr-author-resolves]` **Non-blocking** Nothing gives me more satisfaction that enumerating each specific keys
```suggestion
interface PanelsType extends Record<'floating' | 'normal\>, PanelState> {}
```

File: src/v1/components/surfaces/FloatingPanel/FloatingPanelToggleButton.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  label: string;
+}
+
+const FloatingPanelToggleButton: React.FC<FloatingPanelToggleButtonProps> = ({
+  label,
+  ...props
+}) => {
+  const context = useFloatingPanelContext();
+
+  console.log({ context });
```
Comment (ACoolmanTelicent) [inline src/v1/components/surfaces/FloatingPanel/FloatingPanelToggleButton.tsx:29]:
```suggestion
```

File: src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanels.test.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    try {
+      renderHook(() => useFloatingPanels());
+    } catch (error) {
+      if (error instanceof Error) {
+        expect(error.message).toBe(
+          "FloatingPanel.ToggleButton must be used within a FloatingPanel.Provider"
+        );
+      } else {
+        console.error = originalConsoleError;
+        throw new Error("Test failed because of a unexpected error");
```
Comment (ACoolmanTelicent) [inline src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanels.test.tsx:20]: n1

File: src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanels.test.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    try {
+      renderHook(() => useFloatingPanels());
+    } catch (error) {
+      if (error instanceof Error) {
+        expect(error.message).toBe(
+          "FloatingPanel.ToggleButton must be used within a FloatingPanel.Provider"
+        );
+      } else {
+        console.error = originalConsoleError;
```
Comment (ACoolmanTelicent) [inline src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanels.test.tsx:19]: This should be in a `finally clause`

File: src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanelsContext.test.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { renderHook } from "@testing-library/react";
+import useFloatingPanelContext from "../useFloatingPanelContext";
+
+describe("useFloatingPanels hook", () => {
+  test("throws error when it's not used inside Floating.Provider", () => {
+    // Temporarily suppress console.error
+    const originalConsoleError = console.error;
+    console.error = jest.fn();
```
Comment (ACoolmanTelicent) [inline src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanelsContext.test.ts:8]: What I would do - which I suspect is more robust if we parallelised the tests.

I would do
````
const spies = {
  console: {
    error: jest.spyOn(console, 'error').mockImplementation(() => {})
    }
}

describe('blah', () => {
  beforeEach(() => {
    spies.console.error.resetMocks(); //. ? or is it clearMocks I cant remember
  })
  it('blah', () => {
  })
  ```

File: src/v1/components/surfaces/FloatingPanel/DraggableFloatingPanel.tsx (hunk 1 above)
Comment (AlecsMissangu) [reply src/v1/components/surfaces/FloatingPanel/DraggableFloatingPanel.tsx:47]: It's not a MUI component so I cannot use the sx prop

File: src/v1/components/surfaces/FloatingPanel/FloatingPanelProvider.tsx (hunk 2 above)
Comment (AlecsMissangu) [reply src/v1/components/surfaces/FloatingPanel/FloatingPanelProvider.tsx:8]: The key could be anything. What is your intention behind having them as floating or normal? The current use case would be like 
```
{
  map1: {},
  timeline: {},
  map2: {}
....
}
```

**Review by ACoolmanTelicent — APPROVED** [review]

File: src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanelsContext.test.ts (hunk 6 above)
Comment (ACoolmanTelicent) [reply src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanelsContext.test.ts:8]: feel free to resolve once read

Comment (AlecsMissangu) [reply src/v1/components/surfaces/FloatingPanel/__tests__/useFloatingPanelsContext.test.ts:8]: I don't think I fully understand but I will talk to you about it in a call

**Review by davec504 — APPROVED** [review]

**Pull request merged**